service_identity/src/main/kotlin/org/thechance/service_identity/data/geteway/PermissionGatewayImp.kt

```diff
@@ -0,0 +1,47 @@
+package org.thechance.service_identity.data.geteway
+
+import com.mongodb.client.model.Filters
+import org.bson.types.ObjectId
+import org.litote.kmongo.MongoOperator
+import org.litote.kmongo.eq
+import org.thechance.service_identity.data.DataBaseContainer
+import org.thechance.service_identity.data.collection.AddressCollection
+import org.thechance.service_identity.data.collection.PermissionCollection
+import org.thechance.service_identity.domain.entity.Permission
+import org.thechance.service_identity.domain.gateway.PermissionGateway
+import org.thechance.service_identity.utils.Constants
+
+class PermissionGatewayImp(private val container: DataBaseContainer) : PermissionGateway {
+    private val permissionCollection by lazy {
+        container.database.getCollection<PermissionCollection>(
+            Constants.PERMISSION_COLLECTION_NAME
+        )
+    }
+
+    override suspend fun getPermission(id: String): Permission? {
+        return permissionCollection.findOneById(ObjectId(id))?.toPermission() ?: throw Exception("Wallet not found")
+    }
+
+    override suspend fun addPermission(permission: Permission): Boolean {
+        return permissionCollection.insertOne(permission.toPermissionCollection()).wasAcknowledged()
+
+    }
+
+    override suspend fun deletePermission(permissionId: String): Boolean {
+        return try {
+            val objectId = ObjectId(permissionId)
+            permissionCollection.deleteOne(Filters.eq("_id", objectId))
```

Comment:
use soft delete by using `updateOne `instead of `deleteOne` and filter by isDelete